src/snowflake/connector/aio/_file_transfer_agent.py

```diff
@@ -0,0 +1,308 @@
+        self._source_from_stream = source_from_stream
+        # The list of self-sufficient file metas that are sent to
+        # remote storage clients to get operated on.
+        self._file_metadata: list[SnowflakeFileMeta] = []
+        self._results: list[SnowflakeFileMeta] = []
+        self._multipart_threshold = multipart_threshold or 67108864  # Historical value
+        self._use_s3_regional_url = use_s3_regional_url
+        self._credentials: StorageCredential | None = None
+
+    async def execute(self) -> None:
+        self._parse_command()
+        self._init_file_metadata()
+
+        if self._command_type == CMD_TYPE_UPLOAD:
+            self._process_file_compression_type()
+
+        for m in self._file_metadata:
+            m.sfagent = self
+
+        self._transfer_accelerate_config()
+
+        if self._command_type == CMD_TYPE_DOWNLOAD:
+            if not os.path.isdir(self._local_location):
+                os.makedirs(self._local_location)
+
+        if self._stage_location_type == LOCAL_FS:
+            if not os.path.isdir(self._stage_info["location"]):
+                os.makedirs(self._stage_info["location"])
+
+        for m in self._file_metadata:
+            m.overwrite = self._overwrite
+            m.skip_upload_on_content_match = self._skip_upload_on_content_match
+            m.sfagent = self
+            if self._stage_location_type != LOCAL_FS:
+                m.put_callback = self._put_callback
+                m.put_azure_callback = self._put_azure_callback
+                m.put_callback_output_stream = self._put_callback_output_stream
+                m.get_callback = self._get_callback
+                m.get_azure_callback = self._get_azure_callback
+                m.get_callback_output_stream = self._get_callback_output_stream
+                m.show_progress_bar = self._show_progress_bar
+
+                # multichunk threshold
+                m.multipart_threshold = self._multipart_threshold
+
+        logger.debug(f"parallel=[{self._parallel}]")
+        if self._raise_put_get_error and not self._file_metadata:
+            Error.errorhandler_wrapper(
+                self._cursor.connection,
+                self._cursor,
+                OperationalError,
+                {
+                    "msg": "While getting file(s) there was an error: "
+                    "the file does not exist.",
+                    "errno": ER_FILE_NOT_EXISTS,
+                },
+            )
+        await self.transfer(self._file_metadata)
+
+        # turn enum to string, in order to have backward compatible interface
+
+        for result in self._results:
+            result.result_status = result.result_status.value
+
+    async def transfer(self, metas: list[SnowflakeFileMeta]) -> None:
+        files = [self._create_file_transfer_client(m) for m in metas]
+        is_upload = self._command_type == CMD_TYPE_UPLOAD
+        finish_upload_tasks = []
+        finish_download_tasks = []
```

Comment:
do we need two separate arrays?
looking at the code I think they can be consolidated into one array